File: 400_to_and_fro/scala/src/test/scala/ToandfroSpec.scala
import org.scalatest.Spec
import org.scalatest.matchers.ShouldMatchers._
// import org.spoj.Toandfro

class ToandfroSpec extends Spec {
	describe("Toandfro") {
		it("should decipher string"){
			Toandfro.decipher(5,"toioynnkpheleaigshareconhtomesnlewx") should equal ("theresnoplacelikehomeonasnowynightx")
			Toandfro.decipher(3,"ttyohhieneesiaabss") should equal ("thisistheeasyoneab")
		}
	}
}

File: 400_to_and_fro/ruby/toandfro_spec.rb
$:.unshift(File.expand_path(File.dirname(__FILE__)))

require 'rubygems'
require 'spec'
require 'toandfro'

describe "TO and FRO" do
  it "should decipher sentence" do
    ToAndFro.decipher(5,'toioynnkpheleaigshareconhtomesnlewx').should == 'theresnoplacelikehomeonasnowynightx'
    ToAndFro.decipher(3,'ttyohhieneesiaabss').should == 'thisistheeasyoneab'
  end
end